core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -1652,22 +1679,23 @@ public void testDeleteLogSegmentDueToRetentionSizeBreach(int segmentCount,
         when(mockLog.onlyLocalLogSegmentsSize()).thenReturn(localLogSegmentsSize);
 
         List<RemoteLogSegmentMetadata> segmentMetadataList = listRemoteLogSegmentMetadata(
-                leaderTopicIdPartition, segmentCount, recordsPerSegment, segmentSize, epochEntries);
+            leaderTopicIdPartition, segmentCount, recordsPerSegment, segmentSize, epochEntries, state);
         verifyDeleteLogSegment(segmentMetadataList, deletableSegmentCount, currentLeaderEpoch);
     }
 
     @ParameterizedTest(name = "testDeleteLogSegmentDueToRetentionTimeBreach segmentCount={0} deletableSegmentCount={1}")
-    @CsvSource(value = {"50, 0", "50, 1", "50, 23", "50, 50"})
+    @CsvSource(value = {"50, 0, DELETE_SEGMENT_STARTED", "50, 1, COPY_SEGMENT_FINISHED", "50, 23, DELETE_SEGMENT_STARTED", "50, 50, COPY_SEGMENT_FINISHED"})
```

Comment:
can we skip updating the existing tests? 

There is already a test (testDeleteRetentionMsBeforeSegmentReady) to verify that if the state is DELETE_SEGMENT_FINISHED, no interaction is expected with the RemoteStorageManager.

Reply:
Can we also rename the test?

`testDeleteRetentionMsBeforeSegmentReady` to `testDeleteRetentionMsOnExpiredSegment`  (expired/invalid/something-similar)

Reply:
Agree, applying suggestion